File: src/main/java/me/itzisonn_/meazy/parser/InvalidSyntaxException.java
package me.itzisonn_.meazy.parser;

import me.itzisonn_.meazy.text.TextException;
import me.itzisonn_.meazy.text.Text;
import me.itzisonn_.meazy.text.TextKt;
import org.jspecify.annotations.NullMarked;

@NullMarked
public class InvalidSyntaxException extends TextException {
    public InvalidSyntaxException(int lineNumber, Text text) {
        super(TextKt.translatable("meazy:parser.exception.invalid_syntax", lineNumber, text));
    }
}

File: src/main/kotlin/me/itzisonn_/meazy/parser/InvalidSyntaxException.kt
package me.itzisonn_.meazy.parser

import me.itzisonn_.meazy.util.text.Text
import me.itzisonn_.meazy.util.text.TextException
import me.itzisonn_.meazy.util.text.translatable

class InvalidSyntaxException(lineNumber: Int, text: Text) : TextException(
    translatable("meazy:parser.exception.invalid_syntax", lineNumber, text)
)
